lib/perl/Genome/InstrumentData/AlignmentResult/Command/PicardRnaSeqMetrics.pm

```diff
@@ -78,8 +79,37 @@ sub chart_output_filename {
     return Genome::Model::Build::RnaSeq->default_chart_output_filename;
 }
 
+sub required_files_from_annotation_build {
+    (qw/ rRNA_MT_file annotation_file /);
+}
+
+sub does_annotation_build_have_required_files {
+    my ($self, $annotation_build, $reference_build) = Params::Validate::validate_pos(
+        @_, {isa => __PACKAGE__}, {type => OBJECT}, {type => OBJECT},
+    );
+
+    my @missing_files;
+    for my $file_method ( required_files_from_annotation_build() ) {
+        my $file = $annotation_build->$file_method('gtf', $reference_build->id ,0);
+        if ( not -s $file ) {
+            $file = $annotation_build->$file_method('gtf', undef, 0);
+            if ( not -s $file ) {
+                push @missing_files, $file_method;
+            }
+        }
+    }
+
+    if ( @missing_files ) {
+        die $self->error_message("Cannot proceed! Missing required files from annotation build: @missing_files",);
```

Comment:
I'd prefer that something named `does...` returns true or false, and that something named `verify...`, `validate...`, `ensure...`, etc. either returns true or `die`s.

Since the names of which files are missing are desired, maybe this subroutine should be modeled on `__errors__` where the return value is anything that went wrong?  Then this command can die with the list, and the `RnaSeq` command can skip calling this command without needing to catch an exception.